### ISSUE TITLE
add a canonical way to iterate over all leafs in an SBT

### ISSUE THREAD

**Issue by ctb** [dib-lab/sourmash#349]
should be a simple matter of a generator function..? but would be useful programmatically.

**Comment by ctb** [comment]
it's already there! but it returns a list. we should use a generator instead.

**Comment by luizirber** [comment]
Current:
```python
     def leaves(self):
        return [c for c in self.nodes.values() if isinstance(c, Leaf)]
```

Boring solution:
```python
     def leaves(self):
        for c in self.nodes.values():
            if isinstance(c, Leaf):
                yield c
```

BUT! We have a chance to use `yield from`! (and make the code 3.5+-only compatible =P)
```python
     def leaves(self):
        yield from (c for c in self.nodes.values() if isinstance(c, Leaf))
```
(Don't worry, will probably go with the boring one)

**Comment by ctb** [comment]
fixed in 32b960d2
